diff --git a/media/audio/android/aaudio_stream_wrapper.cc b/media/audio/android/aaudio_stream_wrapper.cc
index 1c671690247e2..85efba89eec8c 100644
--- a/media/audio/android/aaudio_stream_wrapper.cc
+++ b/media/audio/android/aaudio_stream_wrapper.cc
@@ -484,7 +484,7 @@ bool AAudioStreamWrapper::Open() {
     AAudioStreamBuilder_setInputPreset(
         builder, params_.effects() & AudioParameters::ECHO_CANCELLER
                      ? AAUDIO_INPUT_PRESET_VOICE_COMMUNICATION
-                     : AAUDIO_INPUT_PRESET_CAMCORDER);
+                     : AAUDIO_INPUT_PRESET_VOICE_PERFORMANCE);
   }

   // Callbacks
